sdks/java/core/src/main/java/org/apache/beam/sdk/testing/BigqueryMatcher.java

```diff
@@ -151,9 +171,34 @@ public void describeTo(Description description) {
 
   @Override
   public void describeMismatchSafely(PipelineResult pResult, Description description) {
-    description
-        .appendText("was (")
-        .appendText(actualChecksum)
-        .appendText(")");
+    String info;
+    if (response == null) {
+      info = "BigQuery response is null";
+    } else if (response.getRows() == null) {
+      info = "rows that is from BigQuery response is null";
+    } else if (response.getRows().isEmpty()) {
+      info = "rows that is from BigQuery response is empty";
+    } else {
+      info = String.format("was (%s).%n"
+          + "\tTotal number of rows are: %d.%n"
+          + "\tQueried data details:%s",
+          actualChecksum, response.getTotalRows(), printRows(4));
```

Comment:
Pull the 4 out into a constant, also we discussed in person today making this more like 20.